server/src/main/java/org/elasticsearch/index/shard/InternalIndexingStats.java

```diff
@@ -111,20 +111,27 @@ static class StatsHolder {
         private final CounterMetric indexFailed = new CounterMetric();
         private final CounterMetric deleteCurrent = new CounterMetric();
         private final CounterMetric noopUpdates = new CounterMetric();
+        private final long shardRelativeStartTimeInNanos = System.nanoTime();
```

Comment:
This is the time the shard is created locally, not the time it is started. In case of a recovery spanning minutes, I wonder if that could be relevant? Indexing time during recovery is not included AFAICS (which I think is the right way to handle it), but then I think the duration since start should also not include recovery time.

Might be easier to add it as a parameter to the `stats` method and track the time the shard transitions to started (or really `POST_RECOVERY` IIRC) in `IndexShard`?